technical-architecture.md

```diff
@@ -0,0 +1,84 @@
+
+This document describes the technical architecture of Application Service Adapter and its subsystems.
+
+* [Overview](#overview)
+* [High-level architecture](#high-level-architecture)
+* [Experimental Cartographer integration](#cartographer-integration)
+* [Authentication and authorization](#authentication-authorization)
+* [Organization and Space management](#org-space-management)
+* [Building (staging) applications](#building-and-staging-applications)
+* [Services](#services)
+* [Routing](#routing)
+* [App Logging and Metrics](#app-logs-metrics)
+
+## <a id="overview"></a>Overview
+Application Service Adapter implements a subset of the [v3 Cloud Foundry APIs](https://v3-apidocs.cloudfoundry.org/) in order to support common Cloud Foundry developer workflows. Application Service Adapter is installed directly on to a Kubernetes cluster than has Tanzu Application Platform (TAP) installed and provides a CF API translation layer that converts CF API calls into underlying and Kubernetes resources. In addition to this API, Application Service Adapter also provides a set of Kubernetes [custom resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/), [controllers](https://kubernetes.io/docs/concepts/architecture/controller/), and [admission webhooks](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
+
+## <a id="high-level-architecture"></a>High-level architecture
+
+![High-level Application Service Adapter architecture diagram](images/tas-adapter-high-level-architecture.png)
+
+### Components
+Application Service Adapter includes components from the open source Cloud Foundry Korifi project as well as a number of proprietary components for deeper integration with Tanzu Application Platform.
+
+The Korifi components are primarily tasked with implementing the Cloud Foundry data model and APIs. They are:
+
+* **Korifi CRDs:** A set of Kubernetes custom resources under the korifi.cloudfoundry.org Api Group that implement the core set of V3 Cloud Foundry resources. These custom resources replicate the Cloud Foundry data model and are used to store persistent state for the CF API translation layer. These resources can be interacted with directly by existing Kubernetes tooling (e.g. the kubectl CLI) and serve as an extension point for further interoperability.
+* **Korifi API Deployment:** A Golang implementation of a core set of V3 Cloud Foundry APIs that is backed by the Korifi CRDs. Existing Cloud Foundry API clients (such as the cf CLI) can target the Korifi API and continue to use their existing CF developer workflows.
+* **Korifi Controllers Deployment:** A set of Kubernetes controllers that implement CF subsystems by orchestrating and reconciling the Korifi CRDs into consolidated intermediate resources that contain the information necessary to build and run an application. This deployment also runs admission webhooks that validate and normalize Korifi resources.
```

Comment:
Probably we should mention the newer deployments somewhere in here also:
- kpack-image-builder
- statefulset-runner
- job-task-runner
- tas-adapter-telemetry-informer

Reply:
How does this look @julian-hj https://github.com/pivotal/docs-tas-adapter/pull/54/commits/da7693d27819c61fb02816b8cbc3f612565b0853 ?